### PR TITLE
Improve navigation for crud

### PR DESCRIPTION
**Pull request by dtrucs**

### CODE REVIEW

File: src/modules/CRUD/components/Nav/Nav.js [hunk 1]
```diff
@@ -1,22 +1,19 @@
...
 
-import { generateURI } from '../../config';
-import NavIcon from './NavIcon';
+import NavGroup from './NavGroup';
 
 import './styles.scss';
 
 const sortByOrder = ({ order: orderA }, { order: orderB }) =>
-  (orderA !== null ? orderA : 0) - (orderB !== null ? orderB : 0);
+  (orderA !== null ? orderA : Infinity) - (orderB !== null ? orderB : Infinity);
```
Comment (mabhub) [inline src/modules/CRUD/components/Nav/Nav.js]:
```js
const sortBy = prop => ({ [prop]: propA }, { [prop]: propB }) =>
  (propA !== null ? propA : Infinity) - (propB !== null ? propB : Infinity);

const sortByOrder = sortBy('order');
```

File: src/modules/CRUD/components/Nav/Nav.js [hunk 2]
```diff
@@ -1,22 +1,19 @@
...
 
-const getFilteredAndOrderedMenu = menu => menu && (
+const getOrderedMenu = menu => menu && (
```
Comment (mabhub) [inline src/modules/CRUD/components/Nav/Nav.js]: This method may benefit from being memoized.

File: src/modules/CRUD/components/Nav/NavIcon/NavIcon.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Icon } from '@blueprintjs/core';
+
+const NavIcon = ({ src, alt = '', isGroup, ...rest }) =>
+  (src
+    ? (
+      <span {...rest} className="bp3-icon">
```
Comment (mabhub) [inline src/modules/CRUD/components/Nav/NavIcon/NavIcon.js]: You can manage className coming from props by desctructuring it from `props`, then applying with your addition:

```suggestion
      <span {...rest} className={`${className} bp3-icon`}>
```

File: src/modules/CRUD/components/Nav/NavIcon/NavIcon.js [hunk 4]
```diff
@@ -0,0 +1,27 @@
...
+import { Icon } from '@blueprintjs/core';
+
+const NavIcon = ({ src, alt = '', isGroup, className, ...rest }) =>
+  (src
+    ? (
+      <span {...rest} className={`${className} bp3-icon`}>
+        <img className="CRUD-nav__icon" src={src} alt={alt} />
+      </span>
+    )
+    : <Icon {...rest} icon={isGroup ? 'layers' : 'layer'} color="#333" />
```
Comment (mabhub) [inline src/modules/CRUD/components/Nav/NavIcon/NavIcon.js]:
```suggestion
    : <Icon {...rest} className={className} icon={isGroup ? 'layers' : 'layer'} color="#333" />
```

Comment (dtrucs) [reply src/modules/CRUD/components/Nav/NavIcon/NavIcon.js]: I'm not sure for this one the `Icon` component has already its own classNames, i don't want to override them

Comment (mabhub) [reply src/modules/CRUD/components/Nav/NavIcon/NavIcon.js]: You won't override anything: BlueprintJS will merge its own classes and those you provides : https://github.com/palantir/blueprint/blob/1aa71605/packages/core/src/components/icon/icon.tsx#L116

Comment (dtrucs) [reply src/modules/CRUD/components/Nav/NavIcon/NavIcon.js]: Ok, top!

**Pull request merged**